packages/core/admin/_internal/node/createBuildContext.ts

```diff
@@ -170,11 +170,14 @@ const createBuildContext = async ({
 
   const features = strapiInstance.config.get('features', undefined);
 
+  const serveAdminPanel = strapiInstance.config.get('admin.serveAdminPanel', true);
+
   const { bundler = 'webpack', ...restOptions } = options;
 
   const buildContext = {
     appDir,
-    basePath: `${adminPath}/`,
+    // If serveAdminPanel is false, use relative paths for separate deployment
+    basePath: serveAdminPanel ? `${adminPath}/` : '',
```

Comment:
```suggestion
    basePath: serveAdminPanel ? `${adminPath}/` : strapiInstance.config.get('admin.url', ''),
```

That seems like it would match [the documentation](https://docs-v4.strapi.io/dev-docs/admin-panel-customization#different-servers) closer, but I might be missing something.

If we make use of admin.url it would allow users to set whatever prefix they want